pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -239,6 +242,18 @@ func (a *AgentClusterInstall) Load(f asset.FileFetcher) (bool, error) {
 
 	setNetworkType(agentClusterInstall, &types.InstallConfig{}, "NetworkType is not specified in AgentClusterInstall.")
 
+	// Due to OCPBUGS-7495 we previously required lowercase platform names here,
+	// even though that is incorrect. Rewrite to the correct mixed case names
+	// for backward compatibility.
+	switch string(agentClusterInstall.Spec.PlatformType) {
+	case baremetal.Name:
+		agentClusterInstall.Spec.PlatformType = hiveext.BareMetalPlatformType
+	case none.Name:
+		agentClusterInstall.Spec.PlatformType = hiveext.NonePlatformType
+	case vsphere.Name:
+		agentClusterInstall.Spec.PlatformType = hiveext.VSpherePlatformType
+	}
```

Comment:
agentclusterinstall_test.go also needs updating with the new strings, e.g.
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -69,3 +69,3 @@
            	            	   Proxy: (*v1beta1.Proxy)(<nil>),
            	            	-  PlatformType: (v1beta1.PlatformType) (len=9) "baremetal"
            	            	+  PlatformType: (v1beta1.PlatformType) (len=9) "BareMetal"